<?xml version="1.0"?>
<launch>
    <arg name="hand_name"/> 
    <arg name="gazebo" default="false" />
    <arg name="gdb" default="false"/> <!-- for debug with gdb -->
    <arg name="hal_lib" default="DummyHal"/>
    <arg name="matlogger_path" default=""/> <!--/home/pilot/src/ROSEndEffectorPackageManager-->
    <arg name="urdf_path" default="$(find-pkg-prefix end_effector)/configs/urdf/$(var hand_name).urdf"/>
	<arg name="srdf_path" default="$(find-pkg-prefix end_effector)/configs/srdf/$(var hand_name).srdf"/>
    <arg name="actions_folder_path" default="$(env HOME)/ROSEE2/actions/$(var hand_name)/"/>
    
    <node unless="$(var gdb)" exec="UniversalRosEndEffector" name="UniversalRosEndEffector" pkg="end_effector" output="screen">
        <param name="robot_description" value="$(command '$(cat $(var urdf_path))')"/>
        <param name="robot_description_semantic" value="$(command '$(cat $(var srdf_path))')" />
        <param name="urdf_path" value="$(var urdf_path)"/> 
        <param name="srdf_path" value="$(var srdf_path)"/> 
        <param name="actions_folder_path" value="$(var actions_folder_path)"/>
        <param name = "primitive_aggregated_srv_name" value="primitives_aggregated_available" />
        <param name = "selectable_finger_pair_info" value="selectable_finger_pair_info" />
        <param name = "grasping_action_srv_name" value = "grasping_actions_available" />
        <param name = "hand_info" value = "hand_info" />
        <param name = "new_grasping_action_srv_name" value = "new_generic_grasping_action" />

        <!-- param for action server -->
        <param name = "rosAction_grasping_command" value="action_command" />
        <param name="rate" value="100.0"/>
    </node> 

    <!-- the debug mode -->
    <node if="$(var gdb)" exec="UniversalRosEndEffector" name="UniversalRosEndEffector" pkg="end_effector" output="screen" launch-prefix="xterm -e gdb --args">
        <param name="robot_description" value="$(command '$(cat $(var urdf_path))')"/>
        <param name="robot_description_semantic" value="$(command '$(cat $(var srdf_path))')" />
        <param name="urdf_path" value="$(var urdf_path)"/> 
        <param name="srdf_path" value="$(var srdf_path)"/> 
        <param name="actions_folder_path" value="$(var actions_folder_path)"/> 
        <param name = "primitive_aggregated_srv_name" value="primitives_aggregated_available" />
        <param name = "selectable_finger_pair_info" value="selectable_finger_pair_info" />
        <param name = "grasping_action_srv_name" value = "grasping_actions_available" />
        <param name = "hand_info" value = "hand_info" />
        <param name = "new_grasping_action_srv_name" value = "new_generic_grasping_action" />
        
        <!-- param for action server -->
        <param name = "rosAction_grasping_command" value="action_command" />
        <param name="rate" value="100.0"/>
    </node> 

    <include file="$(find-pkg-prefix end_effector)/launch/EEHalExecutor.launch">
        <arg name="gazebo" value="$(var gazebo)"/>
        <arg name="hand_name" value="$(var hand_name)"/> 
        <arg name="hal_lib" value="$(var hal_lib)"/> 
        <arg name="gdb" value="$(var gdb)"/> 
        <arg name="matlogger_path" value="$(var matlogger_path)"/>
    </include>   
     
    <!-- start robot state publisher to transform joint pos into tf for rviz, so it is always on, gazebo or not -->
    <node pkg="robot_state_publisher" exec="robot_state_publisher" name="robot_state_publisher" output="screen" >
        <param name="publish_frequency" type="double" value="200.0" />
        <remap from="joint_states" to="/dummyHal/joint_states" />
    </node> 
    
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find end_effector)/configs/rviz/$(var hand_name).rviz" />
        
</launch>
